src/System.IO.Abstractions.Extensions/IFileSystemExtensions.cs

```diff
@@ -11,5 +11,131 @@ public static IDirectoryInfo CurrentDirectory(this IFileSystem fileSystem)
         {
             return fileSystem.DirectoryInfo.New(fileSystem.Directory.GetCurrentDirectory());
         }
+
+        /// <summary>
+        /// Creates a new <see cref="IDirectoryInfo"/> using a random name from the temp path, and returns an <see cref="IDisposable"/>
+        /// that deletes the directory when disposed.
+        /// </summary>
+        /// <param name="fileSystem">
+        /// The <see cref="IFileSystem"/> in use.
+        /// </param>
+        /// <param name="directoryInfo">
+        /// The <see cref="IDirectoryInfo"/> for the directory that was created.
+        /// </param>
+        /// <returns>
+        /// An <see cref="IDisposable"/> to manage the directory's lifetime.
+        /// </returns>
+        public static IDisposable CreateDisposableDirectory(this IFileSystem fileSystem, out IDirectoryInfo directoryInfo)
+        {
+            return fileSystem.CreateDisposableDirectory(fileSystem.Path.GetRandomTempPath(), out directoryInfo);
+        }
+
+        /// <inheritdoc cref="CreateDisposableDirectory(IFileSystem, out IDirectoryInfo)"/>
+        /// <param name="path">
+        /// The full path to the directory to create.
+        /// </param>
+        /// <exception cref="ArgumentException">
+        /// If the directory already exists.
+        /// </exception>
+        public static IDisposable CreateDisposableDirectory(this IFileSystem fileSystem, string path, out IDirectoryInfo directoryInfo)
```

Comment:
this overload behaves a bit differently than the one it inherits the docs from. probably better to have a dedicated summary